pkg/controller/elasticsearch/client/base.go

```diff
@@ -114,6 +115,9 @@ func (c *baseClient) request(
 	}
 
 	resp, err := c.doRequest(ctx, request)
+	if skipErrFunc != nil && skipErrFunc(err) {
+		err = nil
```

Comment:
I think we should offer an option to print the error, either with a log at the debug level:

```golang
	if err != nil {
		log.V(1).Error(err, "...")
	}
````

or by using the `multierror` package:

```golang
	resp, requestErr := c.doRequest(ctx, request)
	if !skipErrFunc(requestErr) {
		return requestErr
	}

	defer resp.Body.Close()

	if responseObj != nil {
		if err := json.NewDecoder(resp.Body).Decode(responseObj); err != nil {
			return multierror.Append(err, requestErr)
		}
	}
```

Reply:
fc3940e